waiter/integration/waiter/basic_test.clj

```diff
@@ -52,6 +52,27 @@
           (is (= "0" (get-in body-json ["headers" "content-length"])) (str body))
           (is (= "text/plain" (get-in body-json ["headers" "accept"])) (str body))))
 
+      (testing "query-string with special characters"
```

Comment:
We need to do these same tests of the bad query strings with WebSockets.

Reply:
I checked by trying to connect to kitchen directly, the upgrade request goes through fine, but the websocket request is closed when invalid characters are present (e.g. `/hello?q=~!@$%^&*()_-+={}[]|:;'<>,.?&foo=%12jhsdf`).

Reply:
Works for me

Reply:
I was assuming it was going to work with just the current changes, and that all we needed was the tests. If you need to do more work to get the WebSockets working, then I'm fine with delaying that for a later PR.

Reply:
It is hard to write a websocket test with a bad query string with jet/jetty. I get the following error:
```
java.lang.IllegalArgumentException: Illegal character in query at index 38: ws://127.0.0.1:9091/websocket-auth?q=~`!@$%^&*()_-+={}[]|:;'<>,.?&foo=%12jhsdf
```

However, I tried directly using a Chrome plugin and after the Jetty upgrade I get an error console (before it used to just fail in the backend and hang in the Chrome plugin):

<img width="736" alt="screen shot 2018-05-08 at 2 48 53 pm" src="https://user-images.githubusercontent.com/6611249/39782442-d34f09f6-52d7-11e8-83d8-f2dcf62e18ed.png">